Where in the World/Where in the World/MapViewController.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  ViewController.swift
+//  Where in the World
+//
+//  Created by Sheng Xu (TT) on 2/5/20.
+//  Copyright © 2020 Sheng Xu. All rights reserved.
+//
+
+import UIKit
+import MapKit
+import UserNotifications
+
+class MapViewController: UIViewController {
+
+    @IBOutlet var mapView: MKMapView!
+    @IBOutlet var detailView: UIView!
+    @IBOutlet var detailTitleLabel: UILabel!
+    @IBOutlet var detailDescriptionLabel: UILabel!
+    @IBOutlet var detailFavoriteButton: UIButton!
+
+    var places: [Place] = []  // Array of place annotations
+    
+    let id = MKMapViewDefaultAnnotationViewReuseIdentifier
+
+    let dataManagerInstance = DataManager.sharedInstance  // Instance of DataManager
+    
+    // Extra Credit: location manager
+    let locationManager = CLLocationManager()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Request authorization and set the view controller as the location manager's delegate
+        locationManager.requestAlwaysAuthorization()
+        locationManager.delegate = self
+        
+        // Set the view controller as the map view's delegate
+        mapView.delegate = self
+        
+        // Set the map view's properties
+        mapView.showsCompass = false  // Disable the compass
+        mapView.pointOfInterestFilter = .excludingAll  // It does not show any points of intersest
+
+        // Track the user's location on the map
+        mapView.showsUserLocation = true
+        mapView.userTrackingMode = .none
+        
+        // The map’s initial region
+        var initialCoordinate = CLLocationCoordinate2D(latitude: 41.8781, longitude: -87.6298)
+        var span = MKCoordinateSpan(latitudeDelta: 0.15, longitudeDelta: 0.15)
+        // Get region data from Data.plist
```

Comment:
Your `viewDidLoad` became a bit long. Consider moving some logic, such as fetching the locations from Data.plist, into its own function.